cpp/src/parquet/types.cc

```diff
@@ -34,32 +34,44 @@ using arrow::util::Codec;
 
 namespace parquet {
 
-std::unique_ptr<Codec> GetCodecFromArrow(Compression::type codec) {
-  std::unique_ptr<Codec> result;
+namespace {
+arrow::Compression::type TranslateParquetCompressionTypeToArrowCompressionType(
+    Compression::type codec) {
   switch (codec) {
     case Compression::UNCOMPRESSED:
-      break;
+      return ::arrow::Compression::UNCOMPRESSED;
     case Compression::SNAPPY:
-      PARQUET_THROW_NOT_OK(Codec::Create(::arrow::Compression::SNAPPY, &result));
-      break;
+      return ::arrow::Compression::SNAPPY;
     case Compression::GZIP:
-      PARQUET_THROW_NOT_OK(Codec::Create(::arrow::Compression::GZIP, &result));
-      break;
+      return ::arrow::Compression::GZIP;
     case Compression::LZO:
-      PARQUET_THROW_NOT_OK(Codec::Create(::arrow::Compression::LZO, &result));
-      break;
+      return ::arrow::Compression::LZO;
     case Compression::BROTLI:
-      PARQUET_THROW_NOT_OK(Codec::Create(::arrow::Compression::BROTLI, &result));
-      break;
+      return ::arrow::Compression::BROTLI;
     case Compression::LZ4:
-      PARQUET_THROW_NOT_OK(Codec::Create(::arrow::Compression::LZ4, &result));
-      break;
+      return ::arrow::Compression::LZ4;
     case Compression::ZSTD:
-      PARQUET_THROW_NOT_OK(Codec::Create(::arrow::Compression::ZSTD, &result));
-      break;
+      return ::arrow::Compression::ZSTD;
     default:
-      break;
+      throw ParquetException("The Parquet compression type is not recognized.");
   }
+}
+}  // namespace
+
+std::unique_ptr<Codec> GetCodecFromArrow(Compression::type codec) {
+  std::unique_ptr<Codec> result;
+  const arrow::Compression::type compression_type =
+      TranslateParquetCompressionTypeToArrowCompressionType(codec);
+  PARQUET_THROW_NOT_OK(Codec::Create(compression_type, &result));
+  return result;
+}
```

Comment:
It seems like this function is not needed anymore?

Reply:
Removed